phonopy/structure/cells.py

```diff
@@ -457,35 +457,53 @@ def get_equivalent_smallest_vectors(atom_number_supercell,
                                     primitive_lattice,
                                     symprec):
     reduced_bases = get_reduced_bases(supercell.get_cell(), symprec)
-    positions = np.dot(supercell.get_positions(), np.linalg.inv(reduced_bases))
-
-    # Atomic positions are confined into the lattice made of reduced bases.
-    positions -= np.rint(positions)
-
-    p_pos = positions[atom_number_primitive]
-    s_pos = positions[atom_number_supercell]
-
-    # The vector arrow is from the atom in primitive to
-    # the atom in supercell cell plus a supercell lattice
-    # point. This is related to determine the phase
-    # convension when building dynamical matrix.
-    supercell_vectors = np.array([
+    reduced_bases_inv = np.linalg.inv(reduced_bases)
+    cart_positions = supercell.get_positions()
```

Comment:
oh, oops, this still does O(n) work.  I meant to replace this with `get_scaled_positions` and then only multiply the positions we need by the cell.

In any case, I think I'll leave it for now since I don't use any of the functionality that calls this function.